Aikido.Zen.Core/Helpers/LogHelper.cs

```diff
@@ -59,14 +59,30 @@ public static void DebugLog(ILogger logger, string message)
         /// </summary>
         /// <param name="logger">The logger instance to use.</param>
         /// <param name="message">The message to log.</param>
-        public static void ErrorLog(ILogger logger, string message)
+        public static void ErrorLog(ILogger logger, string message) => ErrorLog(logger, null, message);
+
+        /// <summary>
+        /// Logs an error message, after sanitizing the message and applying rate limiting.
+        /// </summary>
+        /// <param name="logger">The logger instance to use.</param>
+        /// <param name="message">The message to log.</param>
+        public static void ErrorLog(ILogger logger, Exception exception, string message)
         {
             // Sanitize the message to prevent log injection
             string sanitizedMessage = SanitizeMessage(message);
-            // we log the message to the outputs defined by the application
-            logger.LogError(sanitizedMessage);
             // we also log the message to the debug output in case the application is running in a debugger
             Console.Error.WriteLine(sanitizedMessage);
+
+            if (exception == null)
+            {
+                // we log the message to the outputs defined by the application
+                logger.LogError(sanitizedMessage);
+            }
+            else
+            {
+                // we log the message to the outputs defined by the application
+                logger.LogError(exception, sanitizedMessage);
```

Comment:
This won't cause too much output in the logs? or why did we add the exception logging here? 🤔

Reply:
Logging exceptions should never actually happen, right :-).

It was added on 2 places (both in `RuntimeAssemblyTracker`):
- When subscribing to the AppDomain ran into an exception
- When processing a new assembly ran into an exception

Runtime sca might not work in these cases, but the application will not have any impact. That's why it's important to still log the exception there, and a utility method was added here.